ProjectEureka/project-eureka-flutter
# PE-86 | Question Archive and Close Service

What I Did:
- Created Service to Close Question Service that sends the ID of the best answer as a body through the URL containing a question ID. Backend will change the "closed" status of the question to True, and the "bestAnswer" status of the selected answer to True.
- Added Close button next to Archive button for the owner of the question in the question details page. This button will redirect to "choose the best answer" page, which is still in the development, and the button "Best Answer" will send Question ID and simulated Answer ID to the close_question service.
- Created service to Archive Question that will change "visible" status to False, and these questions won't appear on home page

How to Test:
- In Mongo or Postman, change the "bestAnswer" boolean of the answer "607ce610930fdd5f952c1ce1" to False, and associated question's "Closed" status to False. Make sure "Visible" status is True
- Go to the More Details of the question, and you must see "Archive" and "Close", which means it assumes you are the owner of the question. For demo purposes, it was changed to True `_moreDetailModel.user.id == currUserId`, however, will be changed back to `!=` later
- Click "Close". It will redirect you to the Choose Best Answer page, which currently only has a demo button "Best Answer". This button must change question status "Closed" to True, and Answer must have "bestAnswer" status True. On the home page, you must see that question no has the status Closed and appears under Active questions
- Now Click the Archive button on the more details page. This button changes the visibility status of the question to False, and it must disappear from the home page
- In both cases, you must see an alert message before going back to home page

NOTE:
- Backend is already updated on EC2
- Make sure Questions on Mongo have "status" attribute changed to "closed"
- If there are NO visible questions in the database, the home page will have a loading circle running continuously, but in reality, it will never happen.
